projectEuler/problem3.js

```diff
@@ -0,0 +1,19 @@
+// prompt: what is the largest prime factor of the number 600851475143
+function rLargestPrime(a, b) {
+  // check if a / b is a whole number
+  if (a % b === 0) {
+    // if a whole number, set a = a / b
+    a = a / b;
+    // if
+    if (a % b !== 0) {
+      return a;
+    } else {
+      return rLargestPrime(a / b, b);
+    }
+  } else {
+    console.log(a, b);
+    return rLargestPrime(a, b + 1);
+  }
+}
```

Comment:
For Project Euler you will need to determine if a number is prime for a lot of problems. I think it would be useful to split this function up into a function such as `isPrime` that returns a boolean value and another function that computes the largest prime factor. This way you can easily re-use`isPrime` code by copy/pasting it into a new problem.

I'm also curious to know how many times this recursive function gets called since I solved this problem without recursion. I think you can obtain that metric by declaring `let counter = 0;` outside the function, adding `counter++;` to the first line inside the function, and then using `console.log(counter);` at the bottom of your file after the function is called.

Reply:
re isPrime: I was thinking about building another function within to check for prime values! Good to know I was on the right path, though I wasn't sure if I should build that within or outside the main function? Not sure what's best practice.

re recursion: I found that when I was trying to construct my for-loops and if-statements, the run time seemed to take forever. Trying out a recursive function, but might go back to the drawing board.

Reply:
Helper functions typically live in the global scope, so outside of the function.

In your future coding endeavours you will find that "helper functions" similar to this can be used hundreds of times throughout a code base in different areas. In these cases, we would define functions like this once and "import" it into the modules that require it. It saves a lot of time and space to create re-usable helper functions if they will get used 😃